### PR TITLE
Extracted processEvent function from screens' run functions

### PR DESCRIPTION
**Pull request by Denmey**

### CODE REVIEW

File: Source/Screens/ScreenBase.hpp [hunk 1]
```diff
@@ -3,10 +3,18 @@
 
 #include <SFML/Graphics.hpp>
 
+const int SCREEN_BASE_NOT_CHANGING_SCREEN = -10;
```
Comment (makometr) [inline Source/Screens/ScreenBase.hpp:6]: translate this variable name, pls

File: Source/Screens/ScreenGame.hpp [hunk 2]
```diff
@@ -12,12 +12,15 @@ class ScreenGame : public ScreenBase {
...
+	// Can't we use settings.getRenderWindow() instead?(@Denmey)
+	int run(sf::RenderWindow &window) override;
 	void drawBackground(sf::RenderWindow &WIN);
-  
+
   protected:
-    const Settings &settings;
-};
\ No newline at end of file
+    Settings &settings; // Deleted const qualifier to be able to use .getRenderWindow() function in processEvent(...).(@Denmey)
```
Comment (makometr) [inline Source/Screens/ScreenGame.hpp:25]: ты бы ещё почту оставлял 💃

File: Source/Screens/ScreenSettings.hpp [hunk 3]
```diff
@@ -7,10 +7,14 @@
 class ScreenSettings : public ScreenBase {
 private:
 	ButtonList buttonList;
+	virtual int processEvent(const sf::Event &event) override;
 public:
 	ScreenSettings(Settings &settings);
-	int run(sf::RenderWindow &App);
+	// TODO:
+	// Why do we need to pass sf::RenderWindow here if we have it in settings?
+	// Can't we use settings.getRenderWindow() instead?(@Denmey)
```
Comment (makometr) [inline Source/Screens/ScreenSettings.hpp:15]: This question for @Transliteration :))

**Review by makometr — APPROVED** [review]
Great! V prodakshn
Change all pr-code again 😢

File: Source/Screens/ScreenBase.hpp (hunk 1 above)
Comment (Denmey) [reply Source/Screens/ScreenBase.hpp:6]: I can't, lol. It's variable that denotes int value that returned when screen is not changed.

Comment (makometr) [reply Source/Screens/ScreenBase.hpp:6]: why -10?
\

Comment (Denmey) [reply Source/Screens/ScreenBase.hpp:6]: -1 and 0 are already used

**Pull request merged**